### PR TITLE
Modernize test tools and apply prettier

### PR DESCRIPTION
**Pull request by staltz**
This PR is in preparation for improving the tests in BIPF and ensuring that we have 100% code coverage. I don't intend to touch the implementation other than code style and things like that.

This one does two things:

- Modernizes test tools
  - Updates tape
  - Adds tap-spec
  - Adds nyc for code coverage
- Applies `prettier` for automatic code style
  - Same style as used in JITDB, ssb-db2 and other recent modules
  - Nothing else significant was done to the JS files

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -314,32 +333,35 @@ function createSeekPath(target) {
...
+      start + varint.decode.bytes + _len
+    ) || target.length - len
+  )
 }
 
 function isNull(tag) {
   return tag === 6
 }
 function isUndefined(tag, firstByte) {
-  return tag === 0xE && firstByte === 2
```
Comment (arj03) [inline index.js:334]: It is common for hex to use capital letters. Is it possible we can keep this?

**Comment by arj03** [comment]
Just found that minor thing, that would be nice to leave as is. Otherwise looks ready to merge. I checked the coverage and it looks good. There is 2 methods not tested, we should maybe check if they are used anywhere and if not remove them. I wonder what seekkey2 is, maybe some attempt at something?

**Comment by staltz** [comment]
Alright, updated the `0xE` thing.

> I checked the coverage and it looks good. There is 2 methods not tested, we should maybe check if they are used anywhere and if not remove them. I wonder what seekkey2 is, maybe some attempt at something?

That'll be the theme of some future PR I'll send.

**Review by arj03 — APPROVED** [review]
Great seeing this module getting updated to the same standard tools we use in the other repos. Have been working great there.

**Pull request merged**